File: src/api/system/log.ts
import { post } from '@/utils/request'

export function getOperationLogList<T>(data: { page: number; pageSize: number; keyword?: string; module?: string; startTime?: number; endTime?: number }) {
  return post<T>({ url: '/log/operationList', data })
}

export function getLoginLogList<T>(data: { page: number; pageSize: number; keyword?: string; startTime?: number; endTime?: number }) {
  return post<T>({ url: '/log/loginList', data })
}

export function clearOperationLog<T>(data?: Record<string, never>) {
  return post<T>({ url: '/log/clearOperation', data: data || {} })
}

export function clearLoginLog<T>(data?: Record<string, never>) {
  return post<T>({ url: '/log/clearLogin', data: data || {} })
}

File: service/router/system/log.go
package system

import (
	"sun-panel/api/api_v1"
	"sun-panel/api/api_v1/middleware"

	"github.com/gin-gonic/gin"
)

func InitLogRouter(router *gin.RouterGroup) {
	api := api_v1.ApiGroupApp.ApiSystem.LogApi

	rAdmin := router.Group("", middleware.LoginInterceptor)
	{
		rAdmin.POST("/log/operationList", middleware.PermissionInterceptor("operation_log:view"), api.GetOperationLogList)
		rAdmin.POST("/log/loginList", middleware.PermissionInterceptor("login_log:view"), api.GetLoginLogList)
		rAdmin.POST("/log/clearOperation", middleware.PermissionInterceptor("operation_log:clear"), api.ClearOperationLog)
		rAdmin.POST("/log/clearLogin", middleware.PermissionInterceptor("login_log:clear"), api.ClearLoginLog)
	}
}